src/handlers/diff_header_misc.rs

```diff
@@ -13,24 +13,18 @@ impl<'a> StateMachine<'a> {
 
     pub fn handle_diff_header_misc_line(&mut self) -> std::io::Result<bool> {
         let is_binary: bool = self.test_diff_is_binary();
-        let file_missing: bool = self.test_diff_file_missing();
 
-        if !file_missing && !is_binary {
+        if !self.test_diff_file_missing() && !is_binary {
             return Ok(false);
         }
 
-        if is_binary {
-            match (self.minus_file.as_str(), self.plus_file.as_str()) {
-                ("", "") => {
-                    return self.handle_additional_cases(match self.state {
-                        State::DiffHeader(_) => self.state.clone(),
-                        _ => State::DiffHeader(DiffType::Unified),
-                    });
-                }
-                ("/dev/null", _) => self.plus_file.push_str(" (binary file)"),
-                (_, "/dev/null") => self.minus_file.push_str(" (binary file)"),
-                (_, _) => (),
-            };
+        if is_binary && !(self.minus_file.is_empty() && self.plus_file.is_empty()) {
+            if self.minus_file != "/dev/null" {
+                self.minus_file.push_str(" (binary file)");
+            }
+            if self.plus_file != "/dev/null" {
+                self.plus_file.push_str(" (binary file)");
+            }
```

Comment:
Here I have originally added the `(binary file)` for the `(_, _)` case for completeness (not strictly related) but then realized the whole `match` could be rewritten in a simpler way and decided to go ahead with that, as it took me a while to understand what this was doing (at first I thought something was wrong here).